gtk/gtk.go

```diff
@@ -9506,6 +9516,254 @@ func (v *TreeRowReference) Valid() bool {
+	obj := glib.Take(unsafe.Pointer(c))
+	return wrapTreeSortable(obj), nil
+}
+
+func wrapTreeSortable(obj *glib.Object) *TreeSortable {
+	return &TreeSortable{obj}
+}
+
+// TreeIterCompareFunc defines the function prototype for the sort function (f arg) for
+// (* TreeSortable).SetSortFunc
+type TreeIterCompareFunc func(model *TreeModel, a, b *TreeIter, userData interface{}) int
+
+// GetSortColumnId() is a wrapper around gtk_tree_sortable_get_sort_column_id().
+func (v *TreeSortable) GetSortColumnId() (int, SortType, bool) {
+	sort := C.toGtkTreeSortable(unsafe.Pointer(v.native()))
+	var column C.gint
+	var order C.GtkSortType
+	ok := gobool(C.gtk_tree_sortable_get_sort_column_id(sort, &column, &order))
+	return int(column), SortType(order), ok
+}
+
+type treeStoreSortFuncData struct {
+	fn       TreeIterCompareFunc
+	userData interface{}
+}
+
+var (
+	treeStoreSortFuncRegistry = struct {
+		sync.RWMutex
+		next int
+		m    map[int]treeStoreSortFuncData
+	}{
+		next: 1,
+		m:    make(map[int]treeStoreSortFuncData),
+	}
+)
+
+// SetSortColumnId() is a wrapper around gtk_tree_sortable_set_sort_column_id().
+func (v *TreeSortable) SetSortColumnId(column int, order SortType) {
+	sort := C.toGtkTreeSortable(unsafe.Pointer(v.native()))
+	C.gtk_tree_sortable_set_sort_column_id(sort, C.gint(column), C.GtkSortType(order))
+}
+
+// SetSortFunc() is a wrapper around gtk_tree_sortable_set_sort_func().
+func (v *TreeSortable) SetSortFunc(sortColumn int, f TreeIterCompareFunc, userData ...interface{}) error {
+	if len(userData) > 1 {
+		return errors.New("userData len must be 0 or 1")
+	}
+
+	t := treeStoreSortFuncData{fn: f}
+	if len(userData) > 0 {
+		t.userData = userData[0]
+	}
+	treeStoreSortFuncRegistry.Lock()
+	id := treeStoreSortFuncRegistry.next
+	treeStoreSortFuncRegistry.next++
+	treeStoreSortFuncRegistry.m[id] = t
+	treeStoreSortFuncRegistry.Unlock()
+
+	C._gtk_tree_sortable_set_sort_func(v.native(), C.gint(sortColumn), C.gpointer(uintptr(id)))
+	return nil
+}
+
+// SetDefaultSortFunc() is a wrapper around gtk_tree_sortable_set_default_sort_func().
+func (v *TreeSortable) SetDefaultSortFunc(f TreeIterCompareFunc, userData ...interface{}) error {
+	if len(userData) > 1 {
+		return errors.New("userData len must be 0 or 1")
+	}
+
+	t := treeStoreSortFuncData{fn: f}
+	if len(userData) > 0 {
+		t.userData = userData[0]
+	}
+	treeStoreSortFuncRegistry.Lock()
+	id := treeStoreSortFuncRegistry.next
+	treeStoreSortFuncRegistry.next++
+	treeStoreSortFuncRegistry.m[id] = t
+	treeStoreSortFuncRegistry.Unlock()
+
+	C._gtk_tree_sortable_set_default_sort_func(v.native(), C.gpointer(uintptr(id)))
+	return nil
+}
+
+func (v *TreeSortable) HasDefaultSortFunc() bool {
+	return gobool(C.gtk_tree_sortable_has_default_sort_func(v.native()))
+}
+
+/*
+ * GtkTreeModelSort
+ */
+
+// TreeModelSort is a representation of GTK's GtkTreeModelSort
+type TreeModelSort struct {
+	*glib.Object
+
+	// Interfaces
+	TreeModel
+	TreeSortable
```

Comment:
Suggestion: rename `TreeSortable` to `ITreeSortable` everywhere in the repository to conform to naming convention.

TreeModel is not really implemented as an interface... the cleanup for that one would take longer